### ISSUE TITLE
Incorrectly updates highlighting on BufEnter, WinEnter etc

### ISSUE THREAD

**Issue by nkgm** [wfxr/minimap.vim#30]
Before the actual problem, thanks for this plugin and your amazing work on `code-minimap` 😄 

## Check list

- [x] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
- [x] I have searched through the existing issues

## Environment info

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Others:

## Version info:

NVIM v0.5.0-587-g0cdd4652d

## Problem and steps to reproduce

When opening/switching buffers, the Minimap window highlights the wrong line. To quickly reproduce, open a fairly large file (>500 loc) and move the cursor to the middle. Minimap should highlight the middle line. Then open a small file (eg 10 loc) in a new split and try switching back and forth between the two. Minimap will highlight the wrong line until the next `CursorMoved` event.

From a quick look, it would appear `s:source_move` is being called prematurely with `getwininfo` returning height info for the previous buffer and not the one currently active.

**Comment by rabirabirara** [comment]
I neglected to comment on this as it wasn't a breaking issue, but in case you're wondering what becomes of this issue, well, the issue is in 

`let mmheight = getwininfo(winid)[0].botline`

in `s:source_move` as you had suggested.  Unfortunately, the function itself seems to be the problem, and there aren't many ways to check the line height of another window.

For example, a fix is to change the line above with

```vimL
execute mmwinnr . 'wincmd w'
let mmheight = line('w$')
execute 'wincmd p'
```
...because that works as intended, but when you benchmark this, it proves to be much slower than using the above one-liner.

```vimL
count  total (s)   self (s)
  665              0.037054     let mmwinnr = bufwinnr('-MINIMAP-')
  665              0.001557     if mmwinnr == -1
                                    return
  665              0.000484     endif
                            
  665              0.002140     let curwinnr = winnr()
  665              0.001099     if curwinnr == mmwinnr
                                    return
  665              0.000378     endif
                            
  665              0.002379     let winid = win_getid(mmwinnr)
  665              0.002105     let curr = line('.') - 1
  665              0.001545     let total = line('$')
  665              0.009053     let mmheight = getwininfo(winid)[0].botline
  665              0.005485     let pos = float2nr(1.0 * curr / total * mmheight) + 1
  665   0.057382   0.010074     call s:highlight_line(winid, pos)
```

```vimL
count  total (s)   self (s)
  665              0.034783     let mmwinnr = bufwinnr('-MINIMAP-')
  665              0.001608     if mmwinnr == -1
                                    return
  665              0.000480     endif
                            
  665              0.001856     if winnr() == mmwinnr
                                    return
  665              0.000437     endif
                            
  665              0.002405     let winid = win_getid(mmwinnr)
  665              0.002053     let curr = line('.') - 1
  665              0.001422     let total = line('$')
                                " let mmheight = getwininfo(winid)[0].botline
  665              0.039377     execute mmwinnr . 'wincmd w'
  665              0.002944     let mmheight = line('w$')
  665              0.013121     execute 'wincmd p'
  665              0.006042     let pos = float2nr(1.0 * curr / total * mmheight) + 1
  665   0.057461   0.009622     call s:highlight_line(winid, pos)
```

(Of course, in normal usage, the speed difference is actually not noticeable.  It's hard to notice it even if you're just holding cursor down, i.e. holding down 'j'.  But it is large, and probably not worth the trade off.)

Someone has actually, coincidentally, submitted an issue on the matter in the Neovim repository.  The issue also exists with Vim, however.   https://github.com/neovim/neovim/issues/13017 is the issue in particular.  This issue will be fixed as soon as that one is.

**Comment by rabirabirara** [comment]
@nkgm Should be fixed by #40.

**Comment by nkgm** [comment]
Thanks @rabirabirara, works beautifully. And many thanks for the extensive write-up 👍
